pkg/kp/rcstore/consul_store.go

```diff
@@ -301,7 +312,17 @@ func (s *consulStore) Watch(rc *fields.RC, quit <-chan struct{}) (<-chan struct{
 	errors := make(chan error)
 	input := make(chan *api.KVPair)
 
-	go consulutil.WatchSingle(kp.RCPath(rc.ID.String()), s.kv, input, quit, errors)
+	go func() {
+		rcp, err := RCPath(rc.ID)
+		if err != nil {
+			select {
+			case <-quit:
+			case errors <- err:
+			}
+			return
```

Comment:
`WatchSingle()` would run until `quit` is closed, then it would close `input` to stop the other goroutine. If you want your error handling to be done inside this goroutine, it should provide the same semantics. Alternatively, you could do the error handling before spawning goroutines (that you know will never produce anything) and provide the expected behavior to the caller of `Watch()` directly.